File: Programmers/level 3/가장 먼 노드.js
function solution(n, edge) {
    let answer = 0;
    let queue=[];
    const graph=Array(n+1).fill(null).map(()=>Array());
    let visited=new Array(n+1).fill(false);
    const dist=new Array(n+1).fill(0);
    edge.forEach(e=>{
        graph[e[0]].push(e[1]);
        graph[e[1]].push(e[0]);
    });
    dist[0]=1;
    const bfs=(v)=>{        
        visited[v]=true;
        queue.push(v);
        while (queue.length){
            let front=queue.shift();
            for (let i=0; i<graph[front].length; i++){
                const next=graph[front][i];
                if (!visited[next]){
                    visited[next]=true;
                    queue.push(next);
                    dist[next]=dist[front]+1;
                }
            }
        }
    }
    bfs(1);
    const max=Math.max(...dist);
    return dist.filter(d=>d===max).length;
}

File: Programmers/level 2/가장 먼 노드.cpp
#include <string>
#include <vector>
#include<queue>
#include <algorithm>

using namespace std;
typedef pair<int, int> pi;
vector<pi> result;

vector<int> bfs(int n, vector<vector<int>> &graph) {    
    vector<bool> visited(n+1, false);
    vector<int> dist(n+1, 0);
    queue<int> q;
    
  	visited[1]=true;
    q.push(1);

    while (!q.empty()) {
        int current=q.front();
        q.pop();
        for (int i=0; i<graph[current].size(); i++) {
        	int next=graph[current][i];
        	if (visited[next]) continue;
        	visited[next]=true;
        	dist[next]=dist[current]+1;
            q.push(next);
        }
        cnt++;
    }
    return dist;
}


int solution(int n, vector<vector<int>> edge) {
    int answer = 0;
    vector<vector<int>> graph(n+1, vector<int>(0));
    for (int i=0; i<edge.size(); i++) {
        graph[edge[i][0]].push_back(edge[i][1]);
        graph[edge[i][1]].push_back(edge[i][0]);
    }
    
	bfs(n, graph);
    
    vector<int> dist=bfs(n, graph);
    
    int max_dist=*max_element(dist.begin(), dist.end());
    
    
    for (int i=0; i<dist.size(); i++) {
        if (dist[i]==max_dist) answer++;
    }
    
    return answer;
}
